src/mokernels/moinput.jl

```diff
@@ -24,10 +24,17 @@ The first `out_dim` elements represent all outputs for the first input, the seco
 
 See [Inputs for Multiple Outputs](@ref) in the docs for more info.
 """
-struct MOInputIsotopicByFeatures{S,T<:AbstractVector{S},Tout_dim<:Integer} <:
-       AbstractVector{Tuple{S,Int}}
+struct MOInputIsotopicByFeatures{
+    S,T<:AbstractVector{S},IdxType,ToutIndices<:AbstractVector{IdxType}
```

Comment:
Wouldn't it be better to use
```suggestion
    S,I,T<:AbstractVector{S},Tout_axis<:AbstractVector{I}
```

Reply:
maybe? Does the order of types make a difference?

Reply:
It matters if you want to dispatch on them. If typically you would only want to dispatch on `S` and `I`, then putting them first is more convenient as you can simply write `::MyType{S,I}`. Otherwise you have to include intermediate type parameters even though they are not used in the function body, e.g., `::MyType{S,A,B,I}` or `::MyType{S,<:SuperTypeOfA,<:SuperTypeOfB,I}`.

Reply:
ah I didn't know you were allowed to use a subset of types